kiwi/repository/apt.py

```diff
@@ -205,8 +205,9 @@ def import_trusted_keys(self, signing_keys):
 
         :param list signing_keys: list of the key files to import
         """
-        for key in signing_keys:
-            self.signing_keys.append(key)
+        if signing_keys:
+            for key in signing_keys:
+                self.signing_keys.append(key)
```

Comment:
JFYI (as is irrelevant for the real fix) extracting elements from an iterator and adding it to a list can be done in Python with `.extend`. So this can be transformed to:

```python
self.signing_keys.extend(signing_keys or [])
```

or if you like this more (I do):

```python
self.signing_keys.extend(signing_keys if signing_keys else [])
```

Reply:
yes I thought about it but the other implementations of the interface contain more code that should not run when no signing_keys exists. Thus I put the decision in the individual method implementation. I realized it is the wrong place because the interface suggests a different method. So you can expect a follow up pull request